resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala

```diff
@@ -136,6 +151,10 @@ class ExecutorPodsAllocator(
 
   def isDeleted(executorId: String): Boolean = deletedExecutorIds.contains(executorId.toLong)
 
+  private[k8s] def stopApplication(exitCode: Int): Unit = {
+    SparkContext.getActive.foreach(_.stop(exitCode))
```

Comment:
It's not elegant, but `AbstractPodsAllocator` is a developer api, introduce parameter sc will change the signature of constructor, suggestions are welcome here.